### ISSUE TITLE
Can't build with vite and SvelteKit 

### ISSUE THREAD

**Issue by infiniteshroom** [prisma/prisma#20757]
### Bug description

Hi. Working on a project at the moment, which uses Prisma from an external node module into another vite based SvelteKit project. This module is just attached using "npm install ../secondModule". To be clear Prisma was generated in the "SecondModule" project and I'd like to be able to use it from there.

Upon "npm start build" I'm getting a bunch of errors like: "TypeError [ERR_INVALID_MODULE_SPECIFIER]: Invalid module ".prisma/client/index" is not a valid package name imported from /firstModule/.svelte-kit/output/server/chunks/hooks.server.js" and "".prisma/client/index" is imported by ".prisma/client/index?commonjs-external", but could not be resolved – treating it as an external dependency."

I've also attempted to log with additional logging/debugging nothing is outputted. 

Any help appreciated on this one.

### How to reproduce

1. Have external module setup with Prisma
2. npm install module into another project
3. Attempt to build project with Vite and SvelteKit.
4. Errors above are outputted and build cannot complete

### Expected behavior

Prisma module should be found

### Prisma information

```prisma
// Add your schema.prisma
```

generator client {
  provider = "prisma-client-js"

}

datasource db {
  provider = "mysql"
  url      = ""
}
model Company {
  id        Int      @id @default(autoincrement())
  name     String
  description   String  @db.Text
  logo      String
  foundedDate DateTime?
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```ts
// Add your code using Prisma Client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient({ log: ['query', 'info', 'warn'] })

export default prisma;

### Environment & setup

- OS: Debian 11
- Database:MySQL
- Node.js version:18.15

### Prisma Version

```5.1.1

```

**Comment by infiniteshroom** [comment]
Believe I've managed to solve this using the below in vite.config.ts. @ sharmapukar217 mentioned this on #20757. Thanks for that. Seem to be a vite issue.

``
		alias: {
		  ".prisma/client/index-browser": "./node_modules/.prisma/client/index-browser.js",
		  ".prisma/client/index": "./../secondModule/node_modules/.prisma/client/index.js",
		  "@prisma/client/runtime/library": "./../secondModule/node_modules/@prisma/client/runtime/library.js",

		}
	  }
});

``

**Comment by jkomyno** [comment]
Hi @infiniteshroom, could you also please paste the output of `prisma -v` from your project, so we know which version of Prisma you're using? Thanks.

**Comment by infiniteshroom** [comment]
Also Just a follow up on this as well. I've moved the Prisma part out of Sveltekit, into a different Node project. It does compile fine, which would have me believe it's either vite or the fact Prisma is include in an external node module. If it could, I'd be keen on merging it back in. So Sveltekit would be a single deployment.

**Comment by CoolBurnX** [comment]
You could try the following:

TLDR; 

Try to create a `.npmrc` file in your root project folder and add this line: ````public-hoist-pattern[]=*prisma*````

-----------------------

I was getting the same error but using turborepo.

I managed to solve it following these guidelines: [Using Prisma with TurboRepo](https://turbo.build/repo/docs/handbook/tools/prisma)

Good luck!

**Comment by janpio** [comment]
Did you see my request for more information or a reproduction project @infiniteshroom?

**Comment by jkomyno** [comment]
Hey @infiniteshroom, could you please reply to @janpio [above](https://github.com/prisma/prisma/issues/20757#issuecomment-1784226210)? Otherwise, without any reproduction on your side, I'm afraid we won't be able to move forward with this issue. Thanks!

**Comment by loky-lp** [comment]
Hi @janpio and @Weakky, i think this will be helpful.

I had the same issue, you can find a reproducible project here [loky-lp/prisma-sveltekit-monorepo-build-error](https://github.com/loky-lp/prisma-sveltekit-monorepo-build-error) (now fixed, consider the first commit)

I managed to solve thanks to the [comment](https://github.com/prisma/prisma/issues/20757#issuecomment-1694511955) of the deleted ccount
> Try to create a `.npmrc` file in your root project folder and add this line: `public-hoist-pattern[]=*prisma*`

- Error: [`bf1db8e`](https://github.com/loky-lp/prisma-sveltekit-monorepo-build-error/tree/bf1db8e000120fe69b8f94247553520884793922)
- Fix: [`109ba20`](https://github.com/loky-lp/prisma-sveltekit-monorepo-build-error/tree/109ba2051b2649bb0c75c5197ff29d2edcd7898c)

I think that this is a problem related to vite and not to prisma, but correct me if i'm wrong
